src/JWT/Builder/JwtBuilder.cs

```diff
@@ -159,6 +159,12 @@ public JwtBuilder WithAlgorithmFactory(IAlgorithmFactory algFactory)
         public JwtBuilder WithAlgorithm(IJwtAlgorithm algorithm)
         {
             _algorithm = algorithm;
+
+            if (String.Equals(algorithm?.Name, nameof(AlgorithmName.None), StringComparison.OrdinalIgnoreCase))

```

Comment:
Do you know if this can be `algorithm is NoneAlgorithm`?

We can add a private ctor to it so there will only `NoneAlgorithm.Instance`.

Reply:
Changing to just _algorithm is NoneAlgorithm works well.

Reply:
I have pushed the simplified check, the singleton Instance is left to do.

Reply:
Singleton is up to you, if it's something you feel would be useful

Reply:
I'll skip the singleton for now and will revisit it in future if needed